### PR TITLE
Fix/m2 3049 show popup with details when item type which is part of conditional was changed

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/shared/hooks/useFilterConditionalLogicByItem.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { ConditionalLogic } from 'shared/state';
+import { getEntityKey } from 'shared/utils';
+import { getItemConditionDependencies } from 'modules/Builder/features/ActivityItems/ActivityItems.utils';
+import { ItemFormValues } from 'modules/Builder/types';
+import { useCurrentActivity } from 'modules/Builder/hooks';
+import { useFormContext } from 'react-hook-form';
+
```
Comment (kdanilenka32) [inline src/shared/hooks/useFilterConditionalLogicByItem.ts:7]: pls move to builder/hooks and fix spaces

Comment (alex-i-e) [reply src/shared/hooks/useFilterConditionalLogicByItem.ts]: done

**Review by muhurov — APPROVED** [review]

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/GroupedSelectSearchController/GroupedSelectSearchController.tsx [hunk 2]
```diff
@@ -76,100 +77,114 @@ export const GroupedSelectSearchController = <T extends FieldValues>({
...
-              value={value}
-              labelId="input-type-label"
-              label={t('inputType')}
-              renderValue={() => (
-                <StyledFlexTopCenter sx={{ maxHeight: '2.3rem' }}>
-                  <StyledFlexTopCenter sx={{ overflow: 'hidden' }}>
-                    <StyledFlexTopCenter sx={{ mr: theme.spacing(1) }}>
-                      {itemsTypeIcons[value]}
+        render={({ field: { onChange, value }, fieldState: { error } }) => {
+          const handleOnSelectChange = (...props: any[]) => {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/GroupedSelectSearchController/GroupedSelectSearchController.tsx:81]: pls don's use any

File: src/modules/Builder/hooks/useFilterConditionalLogicByItem.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { ConditionalLogic } from 'shared/state';
+import { getEntityKey } from 'shared/utils';
+import { getItemConditionDependencies } from 'modules/Builder/features/ActivityItems/ActivityItems.utils';
```
[the remark this answers is not in the document]
Comment (alex-i-e) [reply src/modules/Builder/hooks/useFilterConditionalLogicByItem.ts]: done

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**